Firestore/core/src/model/resource_path.h

```diff
@@ -53,16 +53,14 @@ class ResourcePath : public impl::BasePath<ResourcePath>,
    * Creates and returns a new path from the given resource-path string, where
    * the path segments are separated by a slash "/".
    */
-  static ResourcePath FromString(const std::string& path);
+  static ResourcePath FromString(absl::string_view path);
```

Comment:
Unfortunately, this needs to be reverted.

We previously ran into trouble when using Abseil classes in functions that cross the boundary between the iOS repo and the C++ repo. IIRC, the problem is that Abseil uses inline namespaces so that `absl::string_view` refers to different types between different versions of Abseil. Let's say that a 2019 version of Abseil defines it as `absl::lts_2019_05_01::string_view`, and a 2020 version as `absl::lts_2020_12_12::string_view`.

This becomes a problem in the iOS build because the C++ repo part of our code is compiled with a certain version of Abseil "baked in", whereas the iOS repo part is compiled by our users using the Abseil version that comes from CocoaPods. In all likelihood, these would be two different Abseil versions. So when using `ResourcePath::FromString(absl::string_view)` to communicate, the two parts would see the signature differently; for example:
* the C++ repo part (the public part) would see it as `ResourcePath::FromString(absl::lts_2019_05_01::string_view)`
* the iOS repo part (the core, or internal, part) would see it as `ResourcePath::FromString(absl::lts_2020_12_12::string_view)`.

So the linker would be unable to resolve the dependency.

Reply:
I ... definitely did not know this. Reverted.

Reply:
This mostly applies to `api` and, to lesser extent, `model` (the two most used namespaces by the public code). Unfortunately, we don't have an enforcement mechanism other than looking for this in code reviews. If you have any ideas, let me know.